plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/IcebergNessieTableOperations.java

```diff
@@ -126,7 +128,11 @@ protected void commitToExistingTable(TableMetadata base, TableMetadata metadata)
     {
         verify(version.orElseThrow() >= 0, "commitToExistingTable called on a new table");
         try {
-            nessieClient.commitTable(base, metadata, writeNewMetadata(metadata, version.getAsInt() + 1), table, toKey(new SchemaTableName(database, this.tableName)));
+            if (table == null) {
+                table = nessieClient.table(toIdentifier(new SchemaTableName(database, tableName)));
```

Comment:
can this be a newer version of the table than one we loaded `base` from?
in fact, why `table` can be null for `commitToExistingTable`?

Reply:
Let me check.

Reply:
> in fact, why table can be null for commitToExistingTable

Good question. I found that it was null before this PR also. So, maybe some integration not done properly (from the beginning). I will take a look and handle it separately. I don't want to bloat this PR scope.